component/prometheus/operator/common/crdmanager.go

```diff
@@ -266,21 +301,66 @@ func (c *crdManager) onAddPodMonitor(obj interface{}) {
 }
 func (c *crdManager) onUpdatePodMonitor(oldObj, newObj interface{}) {
 	pm := oldObj.(*promopv1.PodMonitor)
-	c.clearConfigs(pm.Namespace, pm.Name)
+	c.clearConfigs("podMonitor", pm.Namespace, pm.Name)
 	c.addPodMonitor(newObj.(*promopv1.PodMonitor))
 }
+
 func (c *crdManager) onDeletePodMonitor(obj interface{}) {
 	pm := obj.(*promopv1.PodMonitor)
-	c.clearConfigs(pm.Namespace, pm.Name)
+	c.clearConfigs("podMonitor", pm.Namespace, pm.Name)
+	if err := c.apply(); err != nil {
+		level.Error(c.logger).Log("name", pm.Name, "err", err, "msg", "error applying scrape configs after deleting "+c.kind)
+	}
+}
+
+func (c *crdManager) addServiceMonitor(sm *promopv1.ServiceMonitor) {
+	var err error
+	for i, ep := range sm.Spec.Endpoints {
+		var pmc *config.ScrapeConfig
+		pmc, err = c.configGen.GenerateServiceMonitorConfig(sm, ep, i)
+		if err != nil {
+			// TODO(jcreixell): Generate Kubernetes event to inform of this error when running `kubectl get <servicemonitor>`.
```

Comment:
Maybe also fix this TODO? cc @jcreixell

Reply:
This should be a seperate PR. Too complicated to review with all the other changes.